pkg/f1/run/test_runner.go

```diff
@@ -281,13 +280,20 @@ func (r *Run) gatherProgressMetrics(duration time.Duration) {
 	}
 }
 
-func (r *Run) runWorker(input chan int32, stop chan bool, wg *sync.WaitGroup, worker string, workDone chan bool) {
+func (r *Run) runWorker(input <-chan int32, stop <-chan struct{}, wg *sync.WaitGroup, worker string, workDone chan<- bool) {
 	defer wg.Done()
 	for {
 		select {
 		case <-stop:
 			return
 		case iteration := <-input:
+			// if both stop chan is closed and input ch has more iterations,
+			// select will choose a random case. double check if we need to stop
+			select {
+			case <-stop:
+				return
+			default:
+			}
```

Comment:
This will reintroduce the bug that @nicholasham  has been investigating.
I think we need to change this to differentiate stop now (ctrl-c) from stop (max iterations reached). 

Maybe have a stop channel and an interrupt channel - stop will drain the input channel before stopping. Interrupt will work like the current stop and return immediately.